crates/db/src/init.rs

```diff
@@ -9,9 +10,27 @@ static DB: OnceCell<Db> = OnceCell::new();
 pub async fn init(path: &PathBuf) -> Result<Db> {
     let db = Arc::new(DbInner::connect(path).await.unwrap());
     DB.set(db.clone()).unwrap();
+
+    tokio::spawn(async { receiver().await });
+
     Ok(db)
 }
 
 pub fn get() -> Db {
     DB.get().unwrap().clone()
 }
+
+async fn receiver() -> ! {
+    let mut rx = ethui_broadcast::subscribe_internal().await;
+
+    loop {
+        if let Ok(msg) = rx.recv().await {
+            use InternalMsg::*;
+
+            if let NetworkRemoved(chain_id) = msg {
+                let _ = get().remove_contracts(chain_id).await;
+                let _ = get().remove_transactions(chain_id).await;
```

Comment:
there's other tables we can clean from. check other src/queries files. for example: erc20 and native_balances

Reply:
also, you can `get()` a single time and reuse, instead of at every single call